src/quo2/components/share/qr_code/view.cljs

```diff
@@ -1,12 +1,87 @@
 (ns quo2.components.share.qr-code.view
-  (:require [quo2.components.share.qr-code.style :as style]
+  (:require [quo2.components.avatars.account-avatar.view :as account-avatar]
+            [quo2.components.avatars.channel-avatar.view :as channel-avatar]
+            [quo2.components.avatars.user-avatar.view :as user-avatar]
+            [quo2.components.avatars.wallet-user-avatar :as wallet-avatar]
+            [quo2.components.share.qr-code.style :as style]
             [react-native.core :as rn]
-            [react-native.fast-image :as fast-image]))
-
-(defn qr-code
-  [{:keys [source width height]}]
-  [rn/view
-   {:style style/container}
-   [fast-image/fast-image
-    {:source source
-     :style  (style/image width height)}]])
+            [react-native.fast-image :as fast-image]
+            [utils.image-server :as image-server]))
+
+(defn- qr-code-image
+  [{:keys [size url media-server-port]}]
+  (let [media-server-uri (image-server/get-qr-image-uri-for-any-url
+                          {:url         url
+                           :port        media-server-port
+                           :qr-size     (or 400 (int size))
+                           :error-level :highest})]
+    [fast-image/fast-image
+     {:style  (style/qr-image size)
+      :source {:uri media-server-uri}}]))
+
+(defn- avatar-image
+  [{avatar-type :avatar
+    :as         props}]
+  [rn/view {:style style/avatar-overlay}
+   [rn/view
+    {:style (if (= avatar-type :wallet-account)
+              style/avatar-container-rounded
+              style/avatar-container-circular)}
+    (case avatar-type
+      :profile
+      [user-avatar/user-avatar
+       (assoc props
+              :size              :medium
+              :status-indicator? false
+              :online?           false
+              :ring?             false)]
+
+      :wallet-account
+      [account-avatar/view (assoc props :size 48 :type :default)]
+
+      :community
+      [rn/image
+       {:style  style/community-logo-image
+        :source (:picture props)}]
+
+      :channel
+      [channel-avatar/view (assoc props :locked? nil :size :size/l)]
+
+      :saved-address
+      [wallet-avatar/wallet-user-avatar (assoc props :size :large)]
+
+      nil)]])
+
+(defn view
+  "Receives a URL to show a QR code with an avatar (optional) over it.
+   Parameters:
+     - url: String codified in the QR code
+     - media-server-port: receives the value of the subscription `[:mediaserver/port]`
```

Comment:
Sadly, the `media-server-port` is taken from re-frame :cry:  and we need it to ask QR codes from status-go.

I'd really like to remove this parameter, since it is always required, any suggestion?

AFAIK, we are not allowed to perform re-frame calls inside the component library, but in this case, this component requires status-go to generate the QR, so, is it possible to add an exception and avoid throwing a warning?

Reply:
for re frame connections we can make a connected component in status_im2, imo it can still be nice to have the UI handled in quo2 components but we have this wrapped in somewhere like `src/status_im2/common/qr-codes/view`.

Reply:
could also just move the whole component there too ofc

Reply:
@ilmotta 

I really liked @J-Son89 's suggestion, so  this compnent no longer receives `media-server-port` and `url`, instead it receives `qr-image-uri`, which we expect it to be an uri compatible with `fast-image`.

And created  `status-im2.common.qr-codes.view/qr-code` which hgas the `image-server/get-qr-image-uri-for-any-url` call and passes the value to `quo/qr-code`, so in our screens we are just using the common component and we only pass the url string. :+1: 

Thanks for providing alternatives :)